lxd/device/device_utils_network.go

```diff
@@ -560,6 +560,40 @@ func networkSetupHostVethLimits(m deviceConfig.Device) error {
 		}
 	}
 
+	networkPriority := 0
+	if m["limits.priority"] != "" {
+		networkPriority, err = strconv.Atoi(m["limits.priority"])
+		if err != nil {
+			return fmt.Errorf("Failed to parse limits.priority (%s): %s", m["limits.priority"], err)
+		}
+	}
+
+	err = d.state.Firewall.InstanceClearNetPrio(d.inst.Project().Name, d.inst.Name(), veth)
```

Comment:
I'm a little concerned that this will mean that we will make exec calls to iptables/nftables on every NIC start even if they aren't using netprio setting.

Perhaps we should update this function to accept an `oldConfig deviceConfig.Device` argument that represents the old device settings when being updated, or nil if being started, and then we can ascertain whether we need to wipe the old netprio settings before applying the new ones.

In `Update` functions this can be ascertained using `oldConfig := oldDevices[d.name]`, then we'd check for `oldConfig != nil && oldConfig["limits.priority"] != m["limits.priority"]` before clearing the netprio firewall rules.

Reply:
@mihalicyn this still needs addressing.

Reply:
Fixed. Thanks!